### ISSUE TITLE
Windows Docker daemon can not be started by executing dockerd.exe

### ISSUE THREAD

**Issue by akunaatrium** [docker/docker#30240]
**Description**

Docker daemon failed to start for some unknown reason when Windows was restarted. Attempting to start the daemon manually by executing dockerd.exe in an elevated PowerShell gives an error message and does not start.

Also appempting to start the daemon from the Task Manager -> Services does not work. Nothing happens when right clicking on the "docker" service and selecting "Start".

**Steps to reproduce the issue:**
1. Open up an elevated PowerShell session and cd to the location dockerd.exe resides.
2. Execute `.\dockerd.exe`
3. Observe the error message.

**Describe the results you received:**
Error message:
```
Error starting daemon: pid file found, ensure docker is not running or delete C:\ProgramData\docker\docker.pid
```

If the docker.pid file is manually deleted, the command would succeed.

**Describe the results you expected:**
Docker daemon starting normally.

**Additional information you deem important (e.g. issue happens only occasionally):**
Issue happens when the Docker daemon fails to start in Windows startup.

**Output of `docker version`:**

```
Client:
 Version:      1.14.0-dev
 API version:  1.26
 Go version:   go1.7.4
 Git commit:   d1515a5
 Built:        Thu Dec 15 09:24:59 2016
 OS/Arch:      windows/amd64

Server:
 Version:      1.14.0-dev
 API version:  1.26 (minimum version 1.24)
 Go version:   go1.7.4
 Git commit:   d1515a5
 Built:        Thu Dec 15 09:24:59 2016
 OS/Arch:      windows/amd64
 Experimental: false
```

**Output of `docker info`:**

```
Containers: 7
 Running: 7
 Paused: 0
 Stopped: 0
Images: 5
Server Version: 1.14.0-dev
Storage Driver: windowsfilter
 Windows:
Logging Driver: json-file
Plugins:
 Volume: local
 Network: l2bridge l2tunnel nat null overlay transparent
Swarm: inactive
Default Isolation: process
Kernel Version: 10.0 14393 (14393.576.amd64fre.rs1_release_inmarket.161208-2252)
Operating System: Windows Server 2016 Datacenter
OSType: windows
Architecture: x86_64
CPUs: 2
Total Memory: 4 GiB
Name: eti-win-host
ID: MUSO:JQAE:6YYG:RJOD:2V2Y:L5TG:O4JM:LJ7W:PBY4:Y56P:KWAX:RXNQ
Docker Root Dir: C:\ProgramData\docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Azure VM running Windows Server 2016.

**Comment by thaJeztah** [comment]
ping @jstarks @jhowardmsft

**Comment by friism** [comment]
@jstarks @jhowardmsft would it make sense for the Docker daemon to not check for the pid file (when registered with `--register-service`) since The Windows service machinery ensures that there are not multiple instances running, etc. etc.?

**Comment by minida28** [comment]
Hi, I am new to docker. I am exactly facing the same issue.

**Comment by setry14** [comment]
So I figured out what I was doing which led me to this issue which @minida28 posted. I am following the Docker Guide on how to build this image for docker-whale and I tried running the command ` docker build -t docker-whale . ` from command prompt. This gave me the same error. Then I tried running the same command from Docker Quick Start Terminal by going into that directory and it worked correctly. To me it looks like Docker Quick Start Terminal is running on some Linux VM so it establishes a connection whereas if I run commands via Command Prompt on windows it fails with an error.

**Comment by mrthehud** [comment]
@setry14 @minida28 

I'm new to docker on Windows, but I think you need to set up environment variables for docker to work, especially if you're using docker toolbox. You can see what needs to be set by running `docker-machine env default` in your shell.

See also: https://docs.docker.com/toolbox/faqs/troubleshoot/#example-errors

**Comment by williamjablonski** [comment]
I'm getting the same issue.

**Comment by jhowardmsft** [comment]
I'm locking this issue - the original daemon.pid file handling should be fixed when #31668 is merged. If you have a different error, please open new issues.
